### PR TITLE
CE-1431 Classic editor mobile preview should show Mercury

### PR DESCRIPTION
**Pull request by gabrys**

### CODE REVIEW

File: extensions/wikia/EditPreview/EditPreview.i18n.php [hunk 1]
```diff
@@ -3,28 +3,32 @@
...
-	'wikia-editor-preview-venus-width' => 'Venus',
-	'wikia-editor-preview-type-tooltip' => "Changing this option will show you what this article will look like when it's displayed in a browser on small screens, large screens, mobile devices, or your current display.",
 	'wikia-editor-preview-best-practices-notice' => 'Did you know that over 50% of Wikia visitors are reading your content on mobile? Some custom templates may not properly show on mobile devices. To make sure everything looks great to all visitors, follow our best practices.',
 	'wikia-editor-preview-best-practices-button' => 'Best practices',
 	'wikia-editor-preview-best-practices-button-link' => 'http://community.wikia.com/wiki/Help:Wikitext_best_practices',
+	'wikia-editor-preview-current-width' => 'Desktop',
+	'wikia-editor-preview-desc' => 'Enables users to preview their edits before saving them',
+	'wikia-editor-preview-max-width' => 'Desktop XL',
+	'wikia-editor-preview-min-width' => 'Tablet Portrait',
+	'wikia-editor-preview-mercury' => 'Mobile (Mercury)',
```
Comment (hakubo) [inline extensions/wikia/EditPreview/EditPreview.i18n.php]: Mercury is not a 'public' name for new skin - can you talk with some PO on how to name it here?

**Comment by hakubo** [comment]
I don't like that 'wgArticleAsJson' 'leaked' outsied ArticlesApi controller,
but I have no better idea on how to do this right now.
I am just noting to myself that this has to be improved before it gets too wild.

File: extensions/wikia/EditPageLayout/EditPageService.class.php [hunk 2]
```diff
@@ -80,11 +88,12 @@ public function getPreview($wikitext) {
 		 * We still rely on the parser for other stuff
 		 */
 		if ( $this->mTitle->isCssOrJsPage() ) {
-			$html = '<pre>' . htmlspecialchars( $originalWikitext ) . '</pre>';
+			$out = '<pre>' . htmlspecialchars( $originalWikitext ) . '</pre>';
```
Comment (adamkarminski) [inline extensions/wikia/EditPageLayout/EditPageService.class.php]: Check what happens if you try to preview a CSS/JS page.

File: extensions/wikia/EditPageLayout/templates/EditPageLayout_MercuryPreview.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		left: 50%;
+		margin: -8px;
+		position: absolute;
+		top: 50%;
+	}
+	form {
+		display: none;
+	}
+</style>
+<form method="POST" action="<?= htmlspecialchars($mercuryUrl); ?>">
```
Comment (adamkarminski) [inline extensions/wikia/EditPageLayout/templates/EditPageLayout_MercuryPreview.php]: Per discussion think about using ```Sanitizer::encodeAttribute``` here.

File: extensions/wikia/EditPreview/js/preview.js [hunk 4]
```diff
@@ -118,20 +120,14 @@ define('wikia.preview', [
...
+			code: data.html,
+			width: 320,
+			height: 480,
+			scrolling: true
+		});
 
-		doc.open();
-		doc.writeln(data.html);
-		doc.close();
+		$article.html('<div class="mobile-preview"></div>').find('div').html(iframe);
```
Comment (adamkarminski) [inline extensions/wikia/EditPreview/js/preview.js]: Nice!

**Comment by adamkarminski** [comment]
LGTM

**Comment by dianafa** [comment]
can we close this PR? it's been almost a year since it was open and I don't think is valid anymore. Nevertheless, @gabrys, thanks for this great prototype and proof that mercury preview is doable in the limited amount of time and really does work :)

**Comment by tomasznapieralski** [comment]
Stale PR, closing it.

**Pull request closed without merging**